File: inst/tt/tt-template.R
# Load packages -----------------------------------------------------------

library(tidyverse)
library(showtext)
library(ggtext)
library(nrBrand)
library(glue)
library(ggview)


# Load data ---------------------------------------------------------------

tuesdata <- tidytuesdayR::tt_load(date_chr)


# Load fonts --------------------------------------------------------------

font_add_google("Oswald")
font_add_google("Nunito")
showtext_auto()
showtext_opts(dpi = 300)
title_font <- "Oswald"
body_font <- "Nunito"


# Define colours and fonts-------------------------------------------------

bg_col <- "#F2F4F8"
text_col <- "#151C28"
highlight_col <- "#7F055F"


# Data wrangling ----------------------------------------------------------


# Define text -------------------------------------------------------------

social <- nrBrand::social_caption(
  bg_colour = bg_col,
  icon_colour = highlight_col,
  font_colour = text_col,
  font_family = body_font
)
title <- ""
st <- ""
cap <- source_caption(source = "", graphic = social)


# Plot --------------------------------------------------------------------

theme_minimal(base_size = 10, base_family = body_font) +
  theme(
    plot.margin = margin(5, 5, 5, 5),
    plot.title.position = "plot",
    plot.caption.position = "plot",
    plot.background = element_rect(fill = bg_col, colour = bg_col),
    panel.background = element_rect(fill = bg_col, colour = bg_col),
    plot.title = element_textbox_simple(
      colour = text_col,
      hjust = 0,
      halign = 0,
      margin = margin(b = 5, t = 5),
      family = title_font,
      face = "bold",
      size = rel(1.5)
    ),
    plot.subtitle = element_textbox_simple(
      colour = text_col,
      hjust = 0,
      halign = 0,
      margin = margin(b = 5, t = 5),
      family = body_font
    ),
    plot.caption = element_textbox_simple(
      colour = text_col,
      hjust = 0,
      halign = 0,
      margin = margin(b = 0, t = 10),
      family = body_font
    ),
    strip.text = element_textbox_simple(
      face = "bold",
      margin = margin(t = 10),
      size = rel(0.9)
    ),
    panel.grid.minor = element_blank()
  ) +
  canvas(
    width = 7, height = 5,
    units = "in", bg = bg_col,
    dpi = 300
  ) # -> p


# Save --------------------------------------------------------------------

save_ggplot(
  plot = p,
  file = file.path(yr, date_chr, paste0(date_strip, ".png"))
)

File: inst/tt/tt-template.py
import pandas as pd
import plotnine as gg
import matplotlib.pyplot as plt
import matplotlib.font_manager as fm
import PyDyTuesday
import os
import highlight_text as ht


# Load data ---------------------------------------------------------------

PyDyTuesday.get_date(date_chr)
tt_data = pd.read_csv()


# Define colours ----------------------------------------------------------

bg_col = "#F2F4F8"
text_col = "#151C28"
highlight_col = "#7F055F"


# Load fonts --------------------------------------------------------------

oswald_paths = [p for p in fm.findSystemFonts() if 'oswald' in p.lower()]
for p in oswald_paths:
    fm.fontManager.addfont(p)
nunito_paths = [p for p in fm.findSystemFonts() if 'nunito' in p.lower()]
for p in nunito_paths:
    fm.fontManager.addfont(p)
    
available_names = [f.name for f in fm.fontManager.ttflist]
if 'Nunito' in available_names:
    body_font = 'Nunito'
else:
    body_font = 'sans-serif'

if 'Oswald' in available_names:
    title_font = 'Oswald'
else:
    title_font = 'sans-serif'


# Data wrangling ----------------------------------------------------------



# Define text -------------------------------------------------------------

title = ""
st = ""
cap = '<Data::{"fontweight": "bold"}>: source\n<Graphic::{"fontweight": "bold"}>: Nicola Rennie (@nrennie)'


# Plot --------------------------------------------------------------------

p = ()

fig = p.draw()
fig.set_size_inches(7, 5, forward=True)

plt.figure(fig)

ht.fig_text(
    x=0.01, y=0.06, s=cap,
    fontsize=7.5, va='top',
    fontfamily=body_font,
    color=text_col,
    linespacing=0.6,
    fig=fig)
    

# Save --------------------------------------------------------------------

fpath = os.path.join(yr, date_chr, f'{date_strip}.png')
fig.savefig(fpath, dpi=300)
